### DISCUSSION TITLE
Whether its possible to add manually matched points between different images instead of using the SIFT algorithm

### DISCUSSION THREAD

**Discussion by AnvithPabba** [alicevision/Meshroom#2331] [Q&A]
Hi, I am currently trying to reconstruct a 3d image using 4 different cameras (that are stationary) having a decent amount of overlap, but the issue is mainly the cameras are at different heights and views, and SIFT matching algorithms do not work well with the 4 cameras.

To overcome this I have manually annotated 200+ corresponding common points between the 4 camera views and have their coordinates in the COCO json format. (where each point "i" corresponds to the same point in all 4 camera views)

Is there any way I can load these matching points into AliceVision/Meshlab so that it can use these manually annotated common points instead of the ones generated by the SIFT algorithm?

Any help would be much appreciated, thank you!

**Comment by natowi** [comment]
This is not natively supported in Meshroom yet, but take a look at this repository: https://github.com/MrClock8163/MeshroomGCPMarkerAdditions (convert your coco json to csv in the required format to use this node)

Support for unknown image detector/descriptor type is planned for a future release: https://github.com/alicevision/Meshroom/pull/2252 -> https://github.com/alicevision/AliceVision/pull/1598

**Reply by MrClock8163** [reply]
The only downside is that I had to hijack the CCTAG features for this (since that's the most universal of all feature describers, practically just a point), and that only supports marker IDs from 0-127, due to format limitations. So at most 128 unique manual points can be used like this.

**Reply by VMN80** [reply]
I am interested in using this tool, but I am having difficulty with the installation process. Can you please assist me?

**Reply by MrClock8163** [reply]
@VMN80 Which part of it is not clear?

**Reply by VMN80** [reply]
Ok, problem solved.
Now could you tell me how to use more than 128 markers?
Thank you very much
